### ISSUE TITLE
Add PROPFIND support

### ISSUE THREAD

**Issue by asvc** [etesync/etesync-dav#162]
`etesync-dav` returns 500 when client issues `PROPFIND` command:

    HTTP request - OPTIONS http://localhost:37358/USER@example.com/xxxx
    CardDAV - Sending PROPFIND for "current-user-principal"
    HTTP request - PROPFIND http://localhost:37358/USER@example.com/xxxx
    Could not execute request. Server reports error: "HTTP/1.0 500 Internal Server Error"
    CardDAV - no result for "current-user-principal" search
    CardDAV - CardDAV - retrieving Entity Tags for http://localhost:37358/USER@example.com/xxxx
    HTTP request - PROPFIND http://localhost:37358/USER@example.com/xxxx
    CardDAV - ---------------------

Links:
* https://tools.ietf.org/html/rfc4791
* https://github.com/dmfs/davwiki/wiki/PROPFIND
* https://sabre.io/dav/building-a-caldav-client/

Tangentially related to https://github.com/etesync/etesync-dav/issues/161

**Comment by tasn** [comment]
I believe we support propfind (as collection detection works for Evolution, Thunderbird and etc), and etesync-dav should definitely not 500! I guess you are on windows so you can't make it spew out debug output, right?
Which client and what settings are you trying to make it crash?

**Comment by asvc** [comment]
Is there a `--verbose` flag or an exposed logging facility that can export to disk? I am indeed on Windows, but if worse comes to worst, can spin-up a Linux VM for testing.

**Comment by tasn** [comment]
`etesync-dav -D` prints a lot of info to `stderr`, though unfortunately I have no idea how to get that on Windows. :|

**Comment by asvc** [comment]
I don't think it can be passed to `*.exe`. Maybe enable debug output via Flask?

**Comment by tasn** [comment]
Probably easier to just add support for logging to file. I can take a look into that soon. Though maybe try running from Linux first if you are able and see what the error is like?

**Comment by tasn** [comment]
I forgot to comment here, sorry! There's been a version with windows logging for a while now!
https://github.com/etesync/etesync-dav/blob/master/README.md#debugging
